### PR TITLE
Canonicalization metrics improvements

### PR DESCRIPTION
**Pull request by solidoracle**
Follow-up to this https://github.com/paradigmxyz/reth/pull/5454#issuecomment-1836674930

Unified reth_storage_providers_database_duration for now -- if ok will procede with unifying also reth_blockchain_tree_make_canonical_duration as well

### Main Page
Screenshot 2023-12-06 at 15 11 33

### Details Page
Screenshot 2023-12-06 at 15 15 01

### CODE REVIEW

File: etc/grafana/dashboards/overview.json [hunk 1]
```diff
@@ -9788,1235 +9788,61 @@
...
+      {
+        "current": {},
+        "hide": 0,
+        "includeAll": false,
+        "label": "Blockchain Tree Canonicalization Duration Action",
+        "multi": false,
+        "name": "blockchain_tree_action",
+        "options": [],
+        "query": {
+          "query": "reth_blockchain_tree_make_canonical_duration{action=~\".*\"}",
```
Comment (shekhirin) [inline etc/grafana/dashboards/overview.json:9841]: no need to use regex here, just grabbing the label values should work

```json
{
  "current": {},
  "datasource": {
    "type": "prometheus",
    "uid": "${DS_PROMETHEUS}"
  },
  "definition": "label_values(reth_blockchain_tree_make_canonical_duration{instance=\"$instance\"},action)",
  "hide": 0,
  "includeAll": false,
  "label": "Blockchain Tree Canonicalization Action",
  "multi": false,
  "name": "blockchain_tree_action",
  "options": [],
  "query": {
    "query": "label_values(reth_blockchain_tree_make_canonical_duration{instance=\"$instance\"},action)",
    "refId": "PrometheusVariableQueryEditor-VariableQuery"
  },
  "refresh": 1,
  "regex": "",
  "skipUrlSync": false,
  "sort": 0,
  "type": "query"
}
```

image

File: etc/grafana/dashboards/overview.json [hunk 2]
```diff
@@ -9736,9 +9736,9 @@
             "uid": "${DS_PROMETHEUS}"
           },
           "editorMode": "code",
-          "expr": "reth_blockchain_tree_make_canonical_duration{action=\"clone old blocks\", quantile=\"0.5\"} ",
+          "expr": "reth_blockchain_tree_make_canonical_duration{instance=~\"$instance\", action=~\"$blockchain_tree_action\", quantile=\"0.5\"} ",
```
Comment (shekhirin) [inline etc/grafana/dashboards/overview.json:9739]: let's leave only `0.5`, `0.95` and `1` percentile values, so we would get an idea of median, near-maximum and maximum values without too many lines on the chart

**Review by shekhirin — COMMENTED** [review]
Looks good! I don't particularly like the way you need to scroll to the top of the dashboard to change the variable if you aren't in the chart view, but seems like it's not possible to have chart-level variables.

Also, let's put the canonicalization durations under the "Blockchain Tree" section, and the storage durations under the "Database" section.

Just a couple of comments and pending storage durations chart, then should be good to merge!

**Comment by solidoracle** [comment]
@shekhirin thanks for the feedback -- pushed updates

**Review by shekhirin — APPROVED** [review]

**Pull request merged**